ast.go

```diff
@@ -67,10 +69,30 @@ func (*Update) iStatement()         {}
 func (*Grant) iStatement()          {}
 func (*Revoke) iStatement()         {}
 
+// ReadStatementResolver resolves Tableland Custom Functions for a read statement.
+type ReadStatementResolver interface {
```

Comment:
Here's the `ReadStatementResolver` interface that must be implemented in `go-tableland` for reads to pass information to resolve custom functions to values. I opted for the `ReadStatementResolver` name instead of `ReadQueryResolver` to keep things compatible with the parser nomenclature